homework/index.js

```diff
@@ -30,18 +32,116 @@
     return elem;
   }
 
+  // create reusable function.
+  // to remove children of an html element which is (the parent).
+  function removeChildren(element) {
+    while (element.firstChild) {
+      element.removeChild(element.firstChild);
+    }
+  }
+
+  // create left box (repository Info).
+  function createRepoInfo(selectedRepo, repoContainer) {
+    const listRepoInfo = createAndAppend('ul', repoContainer, { class: 'repo-list' });
+
+    const li = createAndAppend('li', listRepoInfo, { text: 'Repository:  ', class: 'li' });
+
+    createAndAppend('a', li, {
+      target: '_blank',
+      href: selectedRepo.html_url,
+      text: selectedRepo.name,
+    });
+
+    createAndAppend('li', listRepoInfo, {
+      text: `Description: ${selectedRepo.description}`,
+      class: 'li',
+    });
+    createAndAppend('li', listRepoInfo, { text: `Forks: ${selectedRepo.forks}`, class: 'li' });
+    createAndAppend('li', listRepoInfo, {
+      text: `Updated: ${selectedRepo.updated_at}`,
+      class: 'li',
+    });
+  }
+
+  // create (right box) as a table to render contributor Info
+  function createContributor(selectedRepo, repoContainer, root) {
+    const contributorUrl = selectedRepo.contributors_url;
+    fetchJSON(contributorUrl).then(contributors => {
+      const table = createAndAppend('table', repoContainer, { class: 'table' });
+
+      const tBody = createAndAppend('tbody', table);
+
+      contributors
+        .forEach(contributor => {
+          const row = createAndAppend('tr', tBody);
+
+          const imageCell = createAndAppend('td', row);
+          createAndAppend('img', imageCell, {
+            src: contributor.avatar_url,
+            class: 'img',
+            alt: 'contributor personal photo',
+          });
+          const nameCell = createAndAppend('td', row);
+          createAndAppend('a', nameCell, {
+            target: '_blank',
+            href: contributor.html_url,
+            text: contributor.login,
+            class: 'contributor-name',
+          });
+          const contributionsNum = createAndAppend('td', row);
+          createAndAppend('span', contributionsNum, {
+            text: contributor.contributions,
+            class: 'contributor-number',
+          });
+        })
+        .catch(err => {
```

Comment:
This line produces an error message in the browser console:

```
Uncaught (in promise) TypeError: Cannot read property 'catch' of undefined
    at index.js:97
```

Reply:
Thanks, U are right! It's done!